src/averageAges.js

```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const onlyMen = century
+    ? people.filter(men => men.sex === 'm'
+      && Math.ceil(men.died / 100) === century)
+    : people.filter(men => men.sex === 'm');
+
+  const averageAge = onlyMen.reduce((sum, men) =>
+    sum + (men.died - men.born), 0) / onlyMen.length;
```

Comment:
don't repeat yourself. check the previous comment. create a separate function and replace all such cases where average age is calculated